app/models/spree/user.rb

```diff
@@ -156,6 +156,14 @@ def disabled=(value)
       self.disabled_at = value == '1' ? Time.zone.now : nil
     end
 
+    def affiliate_enterprises
+      return [] unless Flipper.enabled?(:affiliate_sales_data, self)
+
+      Enterprise.joins(:connected_apps)
+        .where(connected_apps: { type: "ConnectedApps::AffiliateSalesData" })
+        .where.not(connected_apps: { data: nil })
```

Comment:
If we kept one of the old scopes:

```suggestion
      Enterprise.where(id: ConnectedApps::AffiliateSalesData.ready.select(:enterprise_id))
```

Reply:
That looks much more elegant. I like the use of the scope.

Reply:
I've pushed a new commit, using a join to avoid the additional query, let me know what you think.

Reply:
Yes, that's great, too. I don't think it makes a difference to use a join over the `select` which would just do a sub-query in the database. So it wouldn't have been an additional query (I didn't use `pluck`).

Reply:
Ahh, yes I underestimated how clever ActiveRecord is!